### PR TITLE
Add simple peak detection

### PR DESCRIPTION
**Pull request by ChristophWurst**
Fixes https://github.com/nextcloud/suspicious_login/issues/115

### CODE REVIEW

File: lib/Service/LoginClassifier.php [hunk 1]
```diff
@@ -148,13 +150,31 @@ private function persistSuspiciousLogin(string $uid, string $ip) {
...
+		// Assuming that a suspicious IP is most likely one that hasn't been seen before
+		// (for this user), we'll not send another notification until the data is used
+		// for the model training
+		$secondsSinceLastTraining = TrainingDataConfig::default()->getThreshold() * 60 * 60 * 24;
+		if (count($this->mapper->findRelated($uid, $ip, $login, $now - $secondsSinceLastTraining)) > 0) {
+			$this->logger->debug("Notification for $uid:$ip already sent, waiting until the next training period");
+			return;
+		}
+
+		// Have we sent more than 10 alerts in the last week
```
Comment (rullzer) [inline lib/Service/LoginClassifier.php]:
```suggestion
		// Have we sent more than 20 alerts in the last week
```

Comment (rullzer) [reply lib/Service/LoginClassifier.php]: Also this is the last 2 days.... not last week...

File: lib/Service/LoginClassifier.php [hunk 2]
```diff
@@ -148,13 +150,31 @@ private function persistSuspiciousLogin(string $uid, string $ip) {
...
+		$lastTwoDays = count($this->mapper->findRecentByUid($uid, $now - 60 * 60 * 24 * 2));
+		if ($lastTwoDays >= 20) {
+			$this->logger->warning("Suspicious login peak detected: $uid received $lastTwoDays alerts in the last two hours");
```
Comment (rullzer) [inline lib/Service/LoginClassifier.php]: Also doesn't seem correct

File: lib/Service/LoginClassifier.php [hunk 3]
```diff
@@ -148,13 +150,31 @@ private function persistSuspiciousLogin(string $uid, string $ip) {
...
+			return;
+		}
+
+		// Have we sent more than three alerts in the last hour
+		$lastHour = count($this->mapper->findRecentByUid($uid, $now - 60 * 60));
+		if ($lastHour >= 3) {
+			$this->logger->warning("Suspicious login peak detected: $uid received $lastTwoDays alerts in the last hour");
```
Comment (rullzer) [inline lib/Service/LoginClassifier.php]: `$lastTwoDays` seems incorrect

**Comment by ChristophWurst** [comment]
Please have another look.

* Removed the comments as they are prone to get out of sync with the code/notification
* Changed the comparison to `>` as opposed to `>=` as the current alert is included
* Reduced to 10 alerts per two days window max

File: lib/Service/LoginClassifier.php (hunk 3 above)
Comment (rullzer) [reply lib/Service/LoginClassifier.php]:
```suggestion
			$this->logger->warning("Suspicious login peak detected: $uid received $lastHour alerts in the last hour");
```

**Pull request merged**